### PR TITLE
Enhance DataTableColumns ordering options

### PR DESCRIPTION
**Pull request by kimkscott**
This is a partial, draft, PR to enhance DataTableColumns to disallow changing the order of selected columns.

This PR addresses this issue: https://github.com/grommet/grommet/issues/7170

A new attribute, pinOrder, can be specified in an object in the options property.  This attribute pins the order of this column in DataTableColumns.

This is implemented using the existing pinned property of the List component.

Here is a description of requirements provided by Taylor Seamans:
1. Whichever item is effectively the “first” reorderable item should have the up arrow button disabled
2. If an item that isn’t the leftmost is not reorderable, then the items around it should shift while keeping its order fixed
3. This behavior should work for both drag + drop as well as using the arrow keys

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### Screenshots (if appropriate)

image

### CODE REVIEW

File: src/js/components/DataTableColumns/index.d.ts [hunk 1]
```diff
@@ -2,7 +2,15 @@ import * as React from 'react';
...
 export interface DataTableColumnsProps {
   drop: boolean;
-  options: (string | { property: string; label: string; disabled?: boolean })[];
+  options: (
+    | string
+    | {
+        property: string;
+        label: string;
+        disabled?: boolean;
+        pinOrder?: boolean;
```
Comment (britt6612) [inline src/js/components/DataTableColumns/index.d.ts:11]:
```suggestion
        pinned?: boolean;
```

File: src/js/components/DataTableColumns/propTypes.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ if (process.env.NODE_ENV !== 'production') {
         PropTypes.shape({
           disabled: PropTypes.bool,
           label: PropTypes.string,
+          pinOrder: PropTypes.bool,
```
Comment (britt6612) [inline src/js/components/DataTableColumns/propTypes.js:13]:
```suggestion
          pinned: PropTypes.bool,
```

**Comment by britt6612** [comment]
Looking at enhancing/changing current style 

Screen Shot 2024-04-08 at 8 28 26 PM

**Comment by taysea** [comment]
Closing this PR since the work will be continued by https://github.com/grommet/grommet/pull/7193

**Pull request closed without merging**